seedtool/userinterface.ino

```diff
@@ -1397,6 +1491,277 @@ void enter_share() {
+
+          yy = 195; // Absolute, stuck to bottom
+          g_display->setFont(&FreeMono9pt7b);
+          String right_option = "Ok #";
+          int x_r = text_right(right_option.c_str());
+          g_display->setCursor(x_r, yy);
+          g_display->println(right_option);
+
+          String left_option = "Cancel *";
+          g_display->setCursor(0, yy);
+          g_display->println(left_option);
+
+      }
+      while (g_display->nextPage());
+
+      char key;
+      do {
+          key = g_keypad.getKey();
+      } while (key == NO_KEY);
+
+      Serial.println("display xpubs " + String(key));
+      switch (key) {
+        case '#':
+            g_uistate = DISPLAY_XPUBS;
+            return;
+        case '*':
+            g_uistate = SEEDY_MENU;
+            return;
+        case 'A':
+            g_uistate = DERIVATION_PATH;
+            return;
+        case 'B':
+            option_slip132 = !option_slip132;
+            break;
+        case 'C':
+            option_derivation_path = !option_derivation_path;
+            break;
+        default:
+            break;
+      }
+    }
+}
+
+
+void display_xpub(void) {
+    ext_key key;
+    String encoding_type;
+    String derivation_path = keystore.get_derivation_path();
+
+    // @FIXME: return value checks
+    (void)keystore.update(g_master_seed->data, sizeof(g_master_seed->data));
+    (void)keystore.get_xpub(derivation_path.c_str(), &keystore.root, &key);
+
+    char *xpub = NULL;
+    bip32_key_to_base58(&key, BIP32_FLAG_KEY_PUBLIC, &xpub);
+
+    while (true) {
+      g_display->firstPage();
+      do
+      {
+          g_display->setPartialWindow(0, 0, 200, 200);
+          g_display->fillScreen(GxEPD_WHITE);
+          g_display->setTextColor(GxEPD_BLACK);
+
+          const char * title = "Master pubkey";
+          int yy = 18;
+          g_display->setFont(&FreeSansBold9pt7b);
+          Point p = text_center(title);
+          g_display->setCursor(p.x, yy);
+          g_display->println(title);
+
+          switch(xpub_encoding) {
+            case BASE58:
+                g_display->setFont(&FreeMonoBold9pt7b);
+                g_display->setCursor(0, yy + 30);
+                g_display->println(xpub);
+                encoding_type = "Base58";
+                break;
+            case QR_BASE58:
+                displayQR(xpub);
+                encoding_type = "QrBase58";
+                break;
+            case UR:
+                encoding_type = "UR";
+                break;
+            case QR_UR:
+                encoding_type = "QrUR";
+                break;
+            default:
+                break;
+          }
+
+          yy = 195; // Absolute, stuck to bottom
+          g_display->setFont(&FreeMono9pt7b);
+          String right_option = "Back #";
+          int x_r = text_right(right_option.c_str());
+          g_display->setCursor(x_r, yy);
+          g_display->println(right_option);
+
+          String left_option = encoding_type + " *";
```

Comment:
This is a little confusing to me.  Is it telling me the current encoding type?  Or the one that I will see if I press '*'?

Also, everywhere else '*' means "cancel" or "back", it's not consistent here.

Should we add the encoding type option to the prior menu as a selection instead?

Reply:
Fixed